### ISSUE TITLE
[Task] Audit whether I18n domains are handled correctly everywhere

### ISSUE THREAD

**Issue by jrfnl** [rmccue/Requests#758]
Verify whether there are any code paths which could be reached where an international (multi-byte) domain would not have been translated to its ASCII value yet and non-multibyte-safe functions are being used.

This audit should, of course, safeguard this for the future  with additional integration tests if/when needed.

Note: this audit should only verify this for the officially supported API. Not officially supported entry points can be disregarded.

**Comment by schlessera** [comment]
I don't think this only concerns domain handling. I think we'd need to do a full audit of the string-handling in Requests to ensure it handles multi-byte characters gracefully and appropriately everywhere. In some instances, this means properly discarding multi-byte characters upfront, because whatever RFC/standard/protocol disallowing their use.

In some instances, we need to have them properly pass through the string handling without causing invalid characters and random mismatches because of bad truncation of MB characters.

**Comment by jrfnl** [comment]
Also #759 (duplicate)
